pandas/core/groupby/groupby.py

```diff
@@ -2298,7 +2298,18 @@ def levels(self):
 
     @property
     def names(self):
-        return [ping.name for ping in self.groupings]
+        # GH 19029
+        # add suffix to level name in case they contain duplicates (GH 19029):
+        orig_names = [ping.name for ping in self.groupings]
+        # if no names were assigned return the original names
+        if all(x is None for x in orig_names):
+            return orig_names
+        # in case duplicates are contained rename all of them
+        if len(set(orig_names)) < len(orig_names):
+            orig_names = [''.join([str(x), str(i)])
```

Comment:
Use `.format` instead of `.join`. I'd also suggest adding an underscore like we do in merges